src/utils/pathogenParameters.ts

```diff
@@ -0,0 +1,101 @@
+export const pathogenParameters: {
+  [key: string]: {
+    name: string;
+    genomeLength: number | "";
+    subsPerSitePerYear: number | "";
+    serialInterval: number | "";
+  };
+} = {
+  none: {
+    name: "",
+    genomeLength: "",
+    subsPerSitePerYear: "",
+    serialInterval: "",
+  },
+  sarscov2: {
+    name: "SARS-CoV-2",
+    genomeLength: 29903,
+    subsPerSitePerYear: 9.68e-4,
+    serialInterval: 5.2,
+  },
+  monkeypox: {
+    name: "Monkeypox",
+    genomeLength: 197209,
+    subsPerSitePerYear: 5.59e-5,
+    serialInterval: 9.8,
+  },
+  tb: {
+    name: "Tuberculosis",
+    genomeLength: 4411532,
+    subsPerSitePerYear: 1.33e-8,
+    serialInterval: 529,
+  },
+  measles: {
+    name: "Measles",
+    genomeLength: 15894,
+    subsPerSitePerYear: 4.97e-4,
+    serialInterval: 12,
+  },
+  mumps: {
+    name: "Mumps",
+    genomeLength: 15384,
+    subsPerSitePerYear: 4.35e-4,
+    serialInterval: 18,
+  },
+  ebola: {
+    name: "Ebola",
+    genomeLength: 18959,
+    subsPerSitePerYear: 1.05e-3,
+    serialInterval: 13,
+  },
+  other: {
+    name: "Other",
+    genomeLength: "",
+    subsPerSitePerYear: "",
+    serialInterval: "",
+  },
+};
+
+const factorial = (x: number) => {
```

Comment:
Have you considered what happens if `x` comes in but is not a positive integer? I don't know well enough how this is being used, but if you think there's ever the possibility of an upstream mistake resulting in something non-integer getting here, you probably want to have a check that verifies it's an integer and raises an error if not. JS float weirdness could easily get you 2.99999 or something, which is fine, whatever, if `factorial(2.99999)` gets you `5.99995`, that's no big deal. But if you somehow accidentally put in `factorial(2.5)` you probably want it to blow up in your face rather than simply return `3.75`.

Reply:
Good catch, thanks!